### PR TITLE
Closes #622: Apply routine dev dependency updates.

### LINKED ISSUE #622: The dev dependencies need routine updates to satisfy dependabot and re-synchronize with upstream twbs/bootstrap (early December 2022).
This may be the last routine update addressing dependabot issues and upstream changes before splitting the Arizona Bootstrap version 5 development from version 4. The development dependencies between those two are less different than those between the main and v4-dev branches of Twitter Bootstrap, because we've already copied several aspects of main package.json from the main branch there.

### PR DESCRIPTION
**Pull request by mmunro-ltrr**
@babel/core                       ^7.20.2  →   ^7.20.5
@rollup/plugin-babel               ^6.0.2  →    ^6.0.3
@rollup/plugin-commonjs           ^23.0.2  →   ^23.0.3
eslint                            ^8.27.0  →   ^8.28.0
find-unused-sass-variables         ^4.0.4  →    ^4.0.5
hugo-bin                          ^0.93.0  →   ^0.95.0
linkinator                          4.1.0  →     4.1.1
node-sass                          ^7.0.3  →    ^8.0.0
postcss                           ^8.4.18  →   ^8.4.19
postcss-cli                       ^10.0.0  →   ^10.1.0
rollup                            ^2.79.1  →    ^3.5.1
stylelint                        ^14.14.1  →  ^14.15.0
stylelint-config-twbs-bootstrap    ^6.0.0  →    ^7.0.0
terser                             5.15.1  →    5.16.1

Holding back on  eslint-plugin-unicorn (https://github.com/az-digital/arizona-bootstrap/pull/620) until the upstream twbs PR Build(deps-dev): Bump eslint-plugin-unicorn from 44.0.2 to 45.0.1 #37555 is resolved.

### CODE REVIEW

**Comment by mmunro-ltrr** [comment]
This now needs updated with the current version bumps, and there's now a minor change to the upstream Twitter Bootstrap `.eslintrc.json` file to help the eslint-plugin-unicorn update (now merged there).

**Review by joeparsons — APPROVED** [review]

**Comment by joeparsons** [comment]
I'm thinking we may want to hold back on bumping the major version for rollup until we're further along with being compatible with Bootstrap 5.x.

@trackleft noticed some strange `[object Object]` references showing up in the autogenerated banner section of the compiled JS files in this PR branch that I think happening because our `rollup.config.js` file is still more similar to the twbs v4-dev version of that file the twbs main/5.x version of that file has some differences related to the banner script.

(e.g. see https://github.com/twbs/bootstrap/blob/v4-dev/build/rollup.config.js#L36 vs https://github.com/twbs/bootstrap/blob/main/build/rollup.config.js#L43)

**Comment by mmunro-ltrr** [comment]
@joeparsons : I'll back out that change & re-roll the rest.

**Comment by mmunro-ltrr** [comment]
Rather than downgrading rollup, the latest change applies the same fix to `rollup.config.js` that twbs made a few weeks ago. One other upstream change also comes in with the latest commit: twbs is again pinning the version of the terser package to a specific version because newer versions had increased the size of the files it generates. Linting checks were failing for a very obscure reason. There's a navbar example outside the main docs at
https://digital.arizona.edu/arizona-bootstrap/docs/2.0/examples/navbar/
The latest vnu-jar part of the linting checks has taken a dislike to this,  objecting that the document specifies English as the language, but the content is in Catalan (it's really the Latin placeholder text). Changing the text to English kept it happy
https://review.digital.arizona.edu/arizona-bootstrap/issue/622/docs/2.0/examples/navbar/
But we might want to do something else with this generally overlooked demo content.

**Comment by trackleft** [comment]
The Javascript appears to all still work, at least in my browser (Chrome Version 108.0.5359.124).

File: build/rollup.config.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ if (BUNDLE) {
 module.exports = {
   input: path.resolve(__dirname, '../js/src/index.js'),
   output: {
-    banner,
+    banner: banner(),
```
Comment (trackleft) [inline build/rollup.config.js:36]:
```suggestion
    banner: banner(),
    interop: 'amd',
```

Comment (trackleft) [reply build/rollup.config.js:36]: See https://github.com/rollup/plugins/pull/1358

File: package.json [hunk 2]
```diff
@@ -64,35 +64,35 @@
   },
   "devDependencies": {
     "@babel/cli": "^7.19.3",
-    "@babel/core": "^7.20.2",
+    "@babel/core": "^7.20.5",
     "@babel/preset-env": "^7.20.2",
-    "@rollup/plugin-babel": "^6.0.2",
-    "@rollup/plugin-commonjs": "^23.0.2",
+    "@rollup/plugin-babel": "^6.0.3",
+    "@rollup/plugin-commonjs": "^24.0.0",
```
Comment (joeparsons) [inline package.json:70]: I don't think this plugin is actually used in our rollup config.  Looks like twbs/bootstrap only uses it in tests (as far as I can tell).
```suggestion
```

Comment (joeparsons) [reply package.json:70]: FWIW, I think it was me who originally added the rollup plugins (way back in https://github.com/az-digital/arizona-bootstrap/pull/108) and IIRC, I just included all `rollup` packages without doing much detailed investigation about where they were used...

**Comment by mmunro-ltrr** [comment]
Testing the suggestions locally, before pushing any changes....

File: build/rollup.config.js (hunk 1 above)
Comment (mmunro-ltrr) [reply build/rollup.config.js:36]: I'm not seeing this as an option in
https://rollupjs.org/guide/en/#outputinterop
and it does seem to break the build locally.

**Review by cpoff — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**